src/dsl/word/Always.hpp

```diff
@@ -51,35 +59,113 @@ namespace dsl {
+                // thread pool and group as the always reaction
+                auto idle_reaction = std::make_shared<threading::Reaction>(
+                    always_reaction->reactor,
+                    std::vector<std::string>{always_reaction->identifier[0] + " - IDLE Task",
+                                             always_reaction->identifier[1],
+                                             always_reaction->identifier[2],
+                                             always_reaction->identifier[3]},
+                    [always_reaction](threading::Reaction& idle_reaction) -> util::GeneratedCallback {
+                        auto callback = [&idle_reaction, always_reaction](threading::ReactionTask& /*task*/) {
+                            // Get a task for the always reaction and submit it to the scheduler
+                            auto always_task = always_reaction->get_task();
+                            if (always_task) {
+                                always_reaction->reactor.powerplant.submit(std::move(always_task));
                             }
-                        }
-                        catch (...) {
-                        }
-                    }
+
+                            // Get a task for the idle reaction and submit it to the scheduler
+                            auto idle_task = idle_reaction.get_task();
+                            if (idle_task) {
+                                // Set the thread pool on the task
+                                idle_task->thread_pool_descriptor = DSL::pool(*always_reaction);
+
+                                // Make sure that idle reaction always has lower priority than the always reaction
+                                idle_task->priority = DSL::priority(*always_reaction) - 1;
+
+                                // Submit the task to be run
+                                idle_reaction.reactor.powerplant.submit(std::move(idle_task));
+                            }
+                        };
+
+                        // Make sure that idle reaction always has lower priority than the always reaction
+                        return {DSL::priority(*always_reaction) - 1,
+                                DSL::group(*always_reaction),
+                                DSL::pool(*always_reaction),
+                                callback};
+                    });
+
+                // Don't emit stats for the idle reaction
+                idle_reaction->emit_stats = false;
+
+                // Keep this reaction handy so it doesn't go out of scope
+                reaction_store[always_reaction->id] = {always_reaction, idle_reaction};
+
+                // Create an unbinder for the always reaction
+                always_reaction->unbinders.push_back([](threading::Reaction& r) {
+                    r.enabled = false;
+                    reaction_store.erase(r.id);
```

Comment:
I don't disagree about cleaning up the thread pool, but we join on this thread on shutdown and provided the actual reaction behaves properly it will get destroyed like all of the other threads and pool